opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, (list, tuple)) and len(value) > 0:
+            return_code = self._check_sequence(value)
+            if return_code is not None:
+                logger.warning("%s in attribute value sequence", return_code)
+                return
+
         self.attributes[key] = value
 
+    @staticmethod
+    def _check_sequence(sequence: (list, tuple)) -> Optional[str]:
+        """
+        Checks if sequence items are valid and identical
+        """
+        first_element_type = type(sequence[0])
+
+        if issubclass(type(sequence[0]), Number):
+            first_element_type = Number
+
+        for element in sequence:
+
+            if not isinstance(element, (bool, str, Number)):
+                return "invalid type"
+
+            if not isinstance(element, first_element_type):
+                return "different type"
+
+        return None
```

Comment:
nitpick: return None is implicit here.

Reply:
Removing this return statement causes the linter to fail with the error message:

`Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)`

Reply:
Yes, that is specified here: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

_Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)._

Reply:
To clarify, this should be left as `return None` to conform to the style guidelines, correct?